src/associations/belongs-to-many.ts

```diff
@@ -0,0 +1,1316 @@
+    if (options?.through?.where) {
+      throughWhere = {
+        [Op.and]: [throughWhere, options.through.where],
+      };
+    }
+
+    addInclude(findOptions, {
+      association: this.fromTargetToThroughOne,
+      attributes: options?.joinTableAttributes,
+      required: true,
+      paranoid: options?.through?.paranoid ?? true,
+      where: throughWhere,
+    });
+
+    let model = this.target;
+    if (options?.scope != null) {
+      if (!options.scope) {
+        model = model.unscoped();
+      } else if (options.scope !== true) { // 'true' means default scope. Which is the same as not doing anything.
+        model = model.scope(options.scope);
+      }
+    }
+
+    if (options?.schema) {
+      model = model.schema(options.schema, options.schemaDelimiter);
+    }
+
+    return model.findAll(findOptions);
+  }
+
+  /**
+   * Count everything currently associated with this, using an optional where clause.
+   *
+   * @param instance instance
+   * @param options find options
+   */
+  async count(instance: SourceModel, options?: BelongsToManyCountAssociationsMixinOptions<TargetModel>): Promise<number> {
+    const getOptions: BelongsToManyGetAssociationsMixinOptions<TargetModel> = {
+      ...options,
+      attributes: [
+        [fn('COUNT', col([this.target.name, this.targetKeyField].join('.'))), 'count'],
+      ],
+      joinTableAttributes: [],
+      raw: true,
+      plain: true,
+    };
+
+    const result = await this.get(instance, getOptions);
+
+    // @ts-expect-error -- this.get() isn't designed to expect returning a raw output.
+    return Number.parseInt(result.count, 10);
+  }
+
+  /**
+   * Check if one or more instance(s) are associated with this. If a list of instances is passed, the function returns true if _all_ instances are associated
+   *
+   * @param sourceInstance source instance to check for an association with
+   * @param targetInstancesOrPks Can be an array of instances or their primary keys
+   * @param options Options passed to getAssociations
+   */
+  async has(
+    sourceInstance: SourceModel,
+    targetInstancesOrPks: AllowArray<TargetModel | Exclude<TargetModel[TargetKey], any[]>>,
+    options?: BelongsToManyHasAssociationMixinOptions<TargetModel>,
+  ): Promise<boolean> {
+    if (!Array.isArray(targetInstancesOrPks)) {
+      targetInstancesOrPks = [targetInstancesOrPks];
+    }
+
+    const targetPrimaryKeys: Array<TargetModel[TargetKey]> = targetInstancesOrPks.map(instance => {
+      if (instance instanceof this.target) {
+        return (instance as TargetModel).get(this.targetKey);
+      }
+
+      return instance as TargetModel[TargetKey];
+    });
+
+    const associatedObjects: TargetModel[] = await this.get(sourceInstance, {
+      ...options,
+      raw: true,
+      scope: false,
+      attributes: [this.targetKey],
+      joinTableAttributes: [],
+      // TODO: current WhereOptions typings do not allow having 'WhereOptions' inside another 'WhereOptions'
```

Comment:
Just for completeness; I made a comment about this in one of the other files as well. That same comment applies here